src/main/scala/scala/compat/java8/collectionImpl/Stepper.scala

```diff
@@ -0,0 +1,545 @@
+package scala.compat.java8.collectionImpl
+
+import java.util.Spliterator
+
+/** A Stepper is a specialized collection that can step through its
```

Comment:
Normal iterators can't be split effectively.  It's a primitive collection from which you can perform parallel operations.  Spliterator is another idea for how to do it, but it doesn't perform as well due to difficulties in fully inlining the side-effecting closure.